### PR TITLE
Feature/lms 386

### PR DESCRIPTION
**Pull request by HerbertIV**

### CODE REVIEW

File: src/Enum/BadgeType.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace EscolaLms\Courses\Enum;
+
+use EscolaLms\Core\Enums\BasicEnum;
+
+class BadgeType extends BasicEnum
```
Comment (qunabu) [inline src/Enum/BadgeType.php:7]: please remove this code

File: src/Repositories/CourseH5PProgressRepository.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+use EscolaLms\Courses\Repositories\Contracts\CourseH5PProgressRepositoryContract;
+use Illuminate\Contracts\Auth\Authenticatable;
+
+class CourseH5PProgressRepository extends BaseRepository implements CourseH5PProgressRepositoryContract
+{
+    /**
+     * @var array
+     */
+    protected $fieldSearchable = [
+        'quiz_id',
```
Comment (qunabu) [inline src/Repositories/CourseH5PProgressRepository.php:16]: replace all `quiz` keyword with topic  with `topic` to follow names https://github.com/EscolaLMS/Courses/tree/main/doc

File: src/ValueObjects/CourseProgressCollection.php [hunk 3]
```diff
@@ -63,29 +75,70 @@ private function buildProgress(): Collection
         return $progress->sortBy('topic_id')->values();
     }
 
+    public function ping(Topic $topic): self
+    {
+        $progress = $this->courseProgressRepositoryContract->findProgress($topic, $this->user);
+
+        if (($progress->status ?? ProgressStatus::INCOMPLETE) == ProgressStatus::COMPLETE) {
+            throw new RuntimeException("Lecture is already finished.");
```
Comment (qunabu) [inline src/ValueObjects/CourseProgressCollection.php:83]: replace all `lecture` keyword with `lesson`

**Review by qunabu — CHANGES_REQUESTED** [review]

File: src/Enum/BadgeType.php (hunk 1 above)
Comment (qunabu) [reply src/Enum/BadgeType.php:7]: and anything related to `badge`

File: src/ValueObjects/CourseProgressCollection.php (hunk 3 above)
Comment (HerbertIV) [reply src/ValueObjects/CourseProgressCollection.php]: changed

File: src/Repositories/CourseH5PProgressRepository.php (hunk 2 above)
Comment (HerbertIV) [reply src/Repositories/CourseH5PProgressRepository.php]: changed

File: src/Enum/BadgeType.php (hunk 1 above)
Comment (HerbertIV) [reply src/Enum/BadgeType.php]: removed

**Review by qunabu — APPROVED** [review]

**Pull request merged**